mmv1/templates/terraform/examples/secure_source_manager_instance_private_psc_endpoint.tf.erb

```diff
@@ -0,0 +1,145 @@
+  rrdatas = [google_compute_forwarding_rule.fw_rule_service_attachment.ip_address]
+}
+
+resource "google_dns_record_set" "ssm_instance_api_record" {
+  name = "${google_secure_source_manager_instance.<%= ctx[:primary_resource_id] %>.host_config.0.api}."
+  type = "A"
+  ttl = 300
+  managed_zone = google_dns_managed_zone.private_zone.name
+  rrdatas = [google_compute_forwarding_rule.fw_rule_service_attachment.ip_address]
+}
+
+resource "google_dns_record_set" "ssm_instance_git_record" {
+  name = "${google_secure_source_manager_instance.<%= ctx[:primary_resource_id] %>.host_config.0.git_http}."
+  type = "A"
+  ttl = 300
+  managed_zone = google_dns_managed_zone.private_zone.name
+  rrdatas = [google_compute_forwarding_rule.fw_rule_service_attachment.ip_address]
+}
+
+resource "google_privateca_ca_pool" "ca_pool" {
+  name     = "<%= ctx[:vars]['ca_pool_id'] %>"
+  location = "us-central1"
+  tier     = "ENTERPRISE"
+  publishing_options {
+    publish_ca_cert = true
+    publish_crl     = true
+  }
+}
+
+resource "google_privateca_certificate_authority" "root_ca" {
+  pool                     = google_privateca_ca_pool.ca_pool.name
+  certificate_authority_id = "<%= ctx[:vars]['root_ca_id'] %>"
+  location                 = "us-central1"
+  config {
+    subject_config {
+      subject {
+        organization = "google"
+        common_name = "my-certificate-authority"
+      }
+    }
+    x509_config {
+      ca_options {
+        is_ca = true
+      }
+      key_usage {
+        base_key_usage {
+          cert_sign = true
+          crl_sign = true
+        }
+        extended_key_usage {
+          server_auth = true
+        }
+      }
+    }
+  }
+  key_spec {
+    algorithm = "RSA_PKCS1_4096_SHA256"
+  }
+
+  // Disable deletion protections for easier test cleanup purposes
+  deletion_protection = false
+  ignore_active_certificates_on_deletion = true
+  skip_grace_period = true
+}
+
+resource "google_privateca_ca_pool_iam_binding" "ca_pool_binding" {
+  ca_pool = google_privateca_ca_pool.ca_pool.id
+  role = "roles/privateca.certificateRequester"
+
+  members = [
+    "serviceAccount:service-${data.google_project.project.number}@gcp-sa-sourcemanager.iam.gserviceaccount.com"
+  ]
+}
+
+// See https://cloud.google.com/secure-source-manager/docs/create-private-service-connect-instance#root-ca-api
+resource "google_secure_source_manager_instance" "<%= ctx[:primary_resource_id] %>" {
+  instance_id = "<%= ctx[:vars]['instance_id'] %>"
+  location = "us-central1"
+  private_config {
+    is_private = true
+    ca_pool = google_privateca_ca_pool.ca_pool.id
+  }
+  depends_on = [
+    google_privateca_certificate_authority.root_ca,
+    time_sleep.wait_120_seconds
+  ]
+}
+
+# ca pool IAM permissions can take time to propagate
+resource "time_sleep" "wait_120_seconds" {
+  depends_on = [google_privateca_ca_pool_iam_binding.ca_pool_binding]
+
+  create_duration = "120s"
+}
+
+data "google_project" "project" {}
```

Comment:
Similarly, move these to the top of the example.

Reply:
Done